involvedSchemaInfo is invalid for AlterTableMode and RefreshMeta

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
there are no explicit set or schema/table name set
https://github.com/pingcap/tidb/blob/339f07ae8f2ba406abc7f4757ffcce918cc86a56/pkg/ddl/executor.go#L7076-L7084
https://github.com/pingcap/tidb/blob/339f07ae8f2ba406abc7f4757ffcce918cc86a56/pkg/ddl/executor.go#L5769-L5777
```
[2025/07/03 16:44:33.813 +00:00] [ERROR] [wait_group_wrapper.go:178] ["panic in the wait group"] [recover="assert failed, InvolvingSchemaInfo should be like an enumerate type: model.InvolvingSchemaInfo{Database:\"\", Table:\"\", Policy:\"\", ResourceGroup:\"\", Mode:0}"] [stack="github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithLog.func1.1
  pkg/util/wait_group_wrapper.go:178
runtime.gopanic
  GOROOT/src/runtime/panic.go:791
github.com/pingcap/tidb/pkg/util/intest.doPanic
  pkg/util/intest/assert_common.go:58
github.com/pingcap/tidb/pkg/util/intest.doAssert
  pkg/util/intest/assert_common.go:30
github.com/pingcap/tidb/pkg/util/intest.Assert
  pkg/util/intest/assert.go:25
github.com/pingcap/tidb/pkg/ddl.(*runningJobs).checkRunnable
  pkg/ddl/ddl_running_jobs.go:166
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).loadAndDeliverJobs
  pkg/ddl/job_scheduler.go:423
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).schedule
  pkg/ddl/job_scheduler.go:333
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).scheduleLoop
  pkg/ddl/job_scheduler.go:268
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithLog.func1
  pkg/util/wait_group_wrapper.go:181"] [stack="github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithLog.func1.1
  pkg/util/wait_group_wrapper.go:178
runtime.gopanic
  GOROOT/src/runtime/panic.go:791
github.com/pingcap/tidb/pkg/util/intest.doPanic
  pkg/util/intest/assert_common.go:58
github.com/pingcap/tidb/pkg/util/intest.doAssert
  pkg/util/intest/assert_common.go:30
github.com/pingcap/tidb/pkg/util/intest.Assert
  pkg/util/intest/assert.go:25
github.com/pingcap/tidb/pkg/ddl.(*runningJobs).checkRunnable
  pkg/ddl/ddl_running_jobs.go:166
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).loadAndDeliverJobs
  pkg/ddl/job_scheduler.go:423
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).schedule
  pkg/ddl/job_scheduler.go:333
github.com/pingcap/tidb/pkg/ddl.(*jobScheduler).scheduleLoop
  pkg/ddl/job_scheduler.go:268
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithLog.func1
  pkg/util/wait_group_wrapper.go:181"]
```

### 2. What did you expect to see? (Required)
success
### 3. What did you see instead (Required)
assert fail
### 4. What is your TiDB version? (Required)
master
<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/ddl/executor.go
+++ b/pkg/ddl/executor.go
@@ -5776,8 +5776,8 @@ func (e *executor) AlterTableMode(sctx sessionctx.Context, args *model.AlterTabl
 		SQLMode:        sctx.GetSessionVars().SQLMode,
 		InvolvingSchemaInfo: []model.InvolvingSchemaInfo{
 			{
-				Database: schema.Name.O,
-				Table:    table.Meta().Name.O,
+				Database: schema.Name.L,
+				Table:    table.Meta().Name.L,
 			},
 		},
 	}

